contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/performance_mode_controller.py

```diff
@@ -0,0 +1,196 @@
+    PERFORMANCE_FUNCTION_NAME = os.environ.get(
+        "PERFORMANCE_FUNCTION_NAME", "performance_mode"
+    )
+
+    return get_function_from_a_module(
+        module_name=os.path.basename(PERFORMANCE_PYTHON_MODULE_PATH),
+        module_path=PERFORMANCE_PYTHON_MODULE_PATH,
+        function_name=PERFORMANCE_FUNCTION_NAME,
+    )
+
+
+class ModuleLoader:
+    """
+    A class to load Python modules dynamically from a given file path.
+
+    Attributes:
+    -----------
+    module_name : str
+        The name to assign to the module once it's loaded.
+    full_path : str
+        The full file path to the module.
+    module : Optional[object]
+        The loaded module object, initialized to None.
+
+    Methods:
+    --------
+    load_module() -> None:
+        Loads the module from the specified file path.
+    get_function(func_name: str) -> Callable:
+        Retrieves a function from the loaded module.
+    """
+
+    def __init__(self, module_name: str, full_path: str):
+        """
+        Initializes the ModuleLoader with the given module name and path.
+
+        Parameters:
+        -----------
+        module_name : str
+            The name to assign to the module once it's loaded.
+        full_path : str
+            The full file path to the module.
+        """
+        self.module_name = module_name
+        self.full_path = full_path
+        self.module: Optional[object] = None
+
+    def load_module(self) -> None:
+        """
+        Loads the module from the specified file path.
+
+        Raises:
+        -------
+        FileNotFoundError:
+            If the module file does not exist.
+        """
+        try:
+            loader = SourceFileLoader(self.module_name, self.full_path)
+            spec = importlib.util.spec_from_loader(self.module_name, loader)
+            self.module = importlib.util.module_from_spec(spec)
+            loader.exec_module(self.module)
+        except FileNotFoundError:
+            logging.error(
+                "Module '{}' cannot be loaded since '{}' doesn't exist".format(
+                    self.module_name, self.full_path
+                )
+            )
+            raise
+
+    def get_function(self, func_name: str) -> Callable:
+        """
+        Retrieves a function from the loaded module.
+
+        Parameters:
+        -----------
+        func_name : str
+            The name of the function to retrieve.
+
+        Returns:
+        --------
+        Callable:
+            The function object from the module.
+
+        Raises:
+        -------
+        AttributeError:
+            If the function is not found in the module.
+        """
+        if self.module is None:
+            self.load_module()
+        try:
+            return getattr(self.module, func_name)
+        except AttributeError:
+            logging.error(
+                "Function '{}' not found in module '{}'".format(
+                    func_name, self.module_name
+                )
```

Comment:
```suggestion
                "Function '%s' not found in module '%s'",
                func_name,
                self.module_name,
```
per Python documentation, we should append a variable with above format while log with a variable
https://docs.python.org/3/howto/logging.html#logging-variable-data

Reply:
Update